### DISCUSSION TITLE
PHP 8.1 for Debian bookworm

### DISCUSSION THREAD

**Discussion by davidcsterratt** [oerdnj/deb.sury.org#1938] [Q&A]
I over-enthusiastically upgraded from Debian bullseye to bookworm in order to get PHP 8, not realising that Drupal 9 is not compatible with PHP8.2. I looked for PHP 8.1. packages for bookworm on deb.sury.org, but there are  (perhaps unsurprisingly!) no packages for bookworm. I'm wondering if there are any packages I can install from bookworm, or if I am best downgrading to bullseye and installing PHP 8.1 that way.

**Comment by oerdnj** [comment]
Not yet, I will extend the repository to build on bookworm, but it won’t happen overnight.

**Reply by oerdnj** [reply]
That aside, wouldn’t it be just easier to upgrade to Drupal 10?

**Reply by davidcsterratt** [reply]
Thank you for the super-fast and helpful reply.

I have considered upgrading to Drupal 10, but I run several sites on Drupal 9 and looking at just one of them, it appears there are some contributed modules that aren't yet tested with Drupal 10. I have the feeling that upgrading to Drupal 10 could be problematic at this stage (though in 6 months or so it will probably be fine).

Therefore I'm pleased that you're planning to extend the repo to bookworm - that would really help me out of my self-dug hole (and perhaps it would be helpful for others too).
